src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -127,6 +132,16 @@ internal sealed class SolutionRestoreWorker : SolutionEventsListener, ISolutionR
 #endif
                 });
             }
+
+            // Signal the background job runner solution is loaded
+            // Needed when OnAfterBackgroundSolutionLoadComplete fires before
+            // Advise has been called.
+            // IsSolutionFullyLoaded may be expensive so make sure it's called only once.
+            if (!_solutionLoadedTcs.Task.IsCompleted
```

Comment:
I was trying to avoid calling `SolutionManager.IsSolutionFullyLoaded` that can be heavy and expensive when invoked for the first time.